### PR TITLE
Complete rework of conda package generation

### PR DESCRIPTION
**Pull request by hansenms**
This is a complete rework of the conda package generation. There is nothing left of the old implementation, which had not been updated since 2015. 

The package building is controlled with the `build.sh` script at the root:

```text
Builds one or more conda packages.

Usage: ./build.sh [options]

Options:
  -p|--package <package>                 Package to build (defaults to all)
  -u|--user <user>                       Anaconda.org channeluser or organization (default: 'gadgetron')
  -t|--token <token>                     Token for uploading to anaconda.org
  --push                                 Push package to anaconda channel
  --force                                Force push even if package exists  
  -h, --help  Brings up this menu
```

With the updated code, conda packages are built for a number of dependencies (ismrmrd, etc.) required to build the Gadgetron and finally we build a gadgetron package. On push to master (CI), the packages are uploaded (`--push`) to the [Gadgetron Anaconda org](https://anaconda.org/gadgetron/).

This PR is just the first down payment to get the pipeline up and running. More work is needed to build on the mac, supply versions to support multiple versions of dependencies, provide more metadata for the packages, test the packages on install.

### CODE REVIEW

**Comment by hansenms** [comment]
@inati, in case you want to have a look.

**Review by roopchansinghv — COMMENTED** [review]
Are we dropping support for DICOM output?  On GE, since we're not able to send back to the scanner "live" - this is the only "reasonable" output we can expect to get for data from that platform.

Would advise adding 'dcmtk' from BioConda.

**Review by roopchansinghv — APPROVED** [review]

**Review by roopchansinghv — APPROVED** [review]

**Review by roopchansinghv — APPROVED** [review]

**Review by roopchansinghv — APPROVED** [review]

File: ismrmrd-python/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+requirements:
+  build:
+    - numpy
+    - h5py>=2.3
+    - nose>=1.0
+    - xsdata=21.12
+
+  run:
+    - xsdata=21.12
+    - numpy
```
Comment (johnstairs) [inline ismrmrd-python/meta.yaml]: Should we specify a version here (and above)?

File: gadgetron/meta.yaml [hunk 2]
```diff
@@ -1,40 +1,65 @@
...
+    - cuda-libraries
+    - cuda-runtime
+    - cuda-nvcc
+    - fftw=3.3.9
+    - gcc_linux-64=9.4.0
+    - gadgetron-python>=1.3.6
+    - gmock>=1.10.0
+    - gxx_linux-64=9.4.0
+    - gtest=1.10.0
+    - ismrmrd>=1.5.0
```
Comment (johnstairs) [inline gadgetron/meta.yaml]: Should we pin this to the specific version?

**Comment by hansenms** [comment]
@roopchansinghv, I have added dcmtk. We should do that in the Gadgetron devcontainer too, but we can take that as a PR over there once we get this in.

**Review by johnstairs — APPROVED** [review]

**Pull request merged**